pkg/config/config.go

```diff
@@ -30,6 +30,7 @@ type Config struct {
 	KeyFile        string             `yaml:"key_file"`
 	Keys           map[string]string  `yaml:"keys"`
 	LogLevel       string             `yaml:"log_level"`
+	KeyProvider    string             `yaml:"key_provider"`
```

Comment:
Thinking that it'd be more extensible to make this into a struct. It'll make it more flexible in the future to customize the behavior. for example:

```go
type KeyProviderConfig struct {
	// env (default), config, file, redis
	Kind string `yaml:"kind"`
	// used with config and env. we could migrate the top level keys to here.
	Keys map[string]string
	// used with file provider
	Path string
	// key to use as the hash to load keys
	RedisKey string `yaml:"redis_key"`
}
```